### DISCUSSION TITLE
Model Viewer VR

### DISCUSSION THREAD

**Discussion by Lemovision** [google/model-viewer#2941] [General]
Hi is there any plan in the future for VR support in model-viewer with WebXR?
Just like the AR mode when using a smartphone, the AR button appears,
There could be a VR button when using a VR headset such as the Oculus Quest
And then it's possible to walk around the model to examine all details in 6-dof

This already works in some other viewers such as Sketchfab
And here's a bunch of demos which works on the Quest for example
When clicking the VR button it turns it become an immersive experience
This is fully supported in the browser app inside Quest
https://webvr.directory/

I think it's quite amazing for direct browser, and quite more immersive than AR
So it would be great to have for model-viewer since WebXR supports it
I mean the [zuckerverse](https://www.youtube.com/watch?v=b9vWShsmE20) is coming up fast after all :p

![vr](https://user-images.githubusercontent.com/61175245/139560835-50aff7e3-7c6f-4fe6-8d55-bf15f710be0c.png)

**Comment by elalish** [comment]
That would be cool and contributions are welcome! It hasn't been a priority  for us so far and would likely be a lot of work since the interaction model is quite different between phone AR and headset VR. Still, if you have a vision, please  give it a shot  and we'll  be happy to review and merge your PR.

**Reply by Lemovision** [reply]
Ha well I would have no idea how to build this myself, I'm more of a 3d designer than developer
Anyway if you think it's cool maybe give it a try one day when you have some time
Lot of new people buying vr headsets these days

**Reply by sameeryel** [reply]
Agree! As a designer myself, I would love to be able to use the 3D content directly in the Oculus browser.

**Reply by elalish** [reply]
What to throw together a mock of what you think that experience could look like?

**Reply by CaptainEmber** [reply]
A-frame is pretty great, does AR and VR out of the box, but doesn't support IOS because... apple.

**Comment by RichardBradley** [comment]
I think this would be a great addition.

I think the MVP would be for the model to appear in front of you and for you to be able to walk around it in 6DOF and rotate / translate it with the joysticks.

There isn't currently an easy & free way to view 3d models in VR (that I can find) (EDIT: looks like Babylon.js might work for this, although that's a bit more involved / heavyweight than model-viewer)

**Reply by elalish** [reply]
FYI @cabanier

**Reply by CaptainEmber** [reply]
Yeah same as above, Check out A-frame.
